src/System.Net.Http/tests/StressTests/HttpStress/Program.cs

```diff
@@ -168,35 +170,13 @@ void ValidateContent(string expectedContent, string actualContent)
                 Version httpVersion = ctx.GetRandomVersion(httpVersions);
                 (string query, string expected) variables = GetGetQueryParameters(contentSource, ctx, numParameters);
                 using (var req = new HttpRequestMessage(HttpMethod.Get, serverUri + "/variables" + variables.query) { Version = httpVersion })
-                using (HttpResponseMessage m = await ctx.HttpClient.SendAsync(req))
+                using (HttpResponseMessage m = await ctx.HttpClient.SendAsync(req, ctx.CancellationToken))
```

Comment:
Isn't it pretty likely now that cancellation requests will happen after the entire operation has already completed?  That's why the previous version of this was doing the delay and the cancellation inline, prior to actually pulling on the response.

Reply:
Yes, this is likely. It's also likely that cancellation has been triggered before the request has started. I opted to do it like this to keep code within operations simple.

Reply:
> Yes, this is likely...I opted to do it like this to keep code within operations simple.

Functional is better than simple ;)

Reply:
But surely the current approach doesn't eliminate that possibility either?

Reply:
> But surely the current approach doesn't eliminate that possibility either?

Eliminate, no, but it makes it much, much less likely.  For example, if the request content length is greater than the window size and we're not pulling on the response stream, we're guaranteed that the cancellation will happen during the request.

Reply:
Agreed, though the thinking here is we might bump into more issues if we don't strictly order the two events. I would say it probably also mimics "real-word" cancellation scenaria more closely. At the same time, when running the tests I see no evidence of actual cancellations diminishing significantly, though it certainly does happen.

Reply:
> though the thinking here is we might bump into more issues if we don't strictly order the two events

We're going to hit a lot fewer issues if the cancellation generally occurs before the request starts or after the request finishes.

> I would say it probably also mimics "real-word" cancellation scenaria more closely

By far the most common "real-world" cancellation scenario is one where cancellation doesn't occur.  But that's of course not useful for these tests, the purpose of which is to find issues with cancellation occurring within particular portions of the test, e.g. what if cancellation occurs while sending the request body, what if cancellation occurs when receiving the response, etc.

> At the same time, when running the tests I see no evidence of actual cancellations diminishing significantly, though it certainly does happen.

If that's true that's surprising to me.  That suggests most of the requests are taking at least ~15ms, on loopback, which means at best we're only getting ~60 requests / second.  If that's true, that's not good.  I'd also question whether the cancellation occurring is actually happening at a point in time that's meaningful, rather than, for example, just hitting a check like this one:
https://github.com/dotnet/corefx/blob/a7efd2a2b3c638b5490547ad6beca128ba5d7bfe/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs#L897-L900

Reply:
@stephentoub as discussed, I've updated the logic so that cancellation is managed from a wrapper function.